src/main/java/org/apache/maven/plugins/gpg/GpgVersion.java

```diff
@@ -32,54 +33,46 @@
  */
 public class GpgVersion implements Comparable<GpgVersion>
 {
-    private final String rawVersion;
 
-    private GpgVersion( String rawVersion )
-    {
-        this.rawVersion = rawVersion;
-    }
+    private static final Pattern VERSION_PATTERN = Pattern.compile( "(\\d+\\.)+(\\d+)" );
 
-    public static GpgVersion parse( String rawVersion )
+    private final int[] versionSegments;
+
+    private GpgVersion( int... versionSegments )
     {
-        return new GpgVersion( rawVersion );
+        this.versionSegments = versionSegments;
     }
 
-    @Override
-    public int compareTo( GpgVersion other )
+    public static GpgVersion parse( String rawVersion )
     {
-        Pattern p = Pattern.compile( "(\\d+\\.)+(\\d+)" );
-
-        String[] thisSegments;
-        Matcher m = p.matcher( rawVersion );
-        if ( m.find() )
-        {
-            thisSegments  = m.group( 0 ).split( "\\." );
-        }
-        else
+        final Matcher versionMatcher = VERSION_PATTERN.matcher( rawVersion );
+        if ( !versionMatcher.find() )
         {
-          throw new IllegalArgumentException( "Can't parse version of " + this.rawVersion );
+            throw new IllegalArgumentException( "Can't parse version of " + rawVersion );
         }
 
-        String[] otherSegments;
-        m = p.matcher( other.rawVersion );
-        if ( m.find() )
-        {
-            otherSegments  = m.group( 0 ).split( "\\." );
-        }
-        else
+        final String[] rawVersionSegments = versionMatcher.group( 0 ).split( "\\." );
+
+        final int[] versionSegments = new int[rawVersionSegments.length];
+        for ( int index = 0; index < rawVersionSegments.length; index++ )
         {
-          throw new IllegalArgumentException( "Can't parse version of " + other.rawVersion );
+            versionSegments[index] = Integer.parseInt( rawVersionSegments[index] );
         }
 
+        return new GpgVersion( versionSegments );
+    }
+
+    @Override
+    public int compareTo( GpgVersion other )
+    {
+        final int[] thisSegments = versionSegments;
+        final int[] otherSegments = other.versionSegments;
+
         int minSegments = Math.min( thisSegments.length, otherSegments.length );
 
         for ( int index = 0; index < minSegments; index++ )
         {
-            int thisValue = Integer.parseInt( thisSegments[index] );
-
-            int otherValue = Integer.parseInt( otherSegments[index] );
-
-            int compareValue = Integer.compare( thisValue, otherValue );
+            int compareValue = Integer.compare( thisSegments[index], otherSegments[index] );
```

Comment:
can this be a simple int comparison? e.g.

```
if (thisSegments[index] != otherSegments[index]) {
  return thisSegments[index] - otherSegments[index]
}
```

(Double check my order there)

Reply:
I think it would be preferable to adhere to the contract of Integer#compareTo / Integer#compare, which is internally implemented as `(x < y) ? -1 : ((x == y) ? 0 : 1)`, because this class is in my opinion just a container for an int array and should behave as such.

But your solution works of course, too. Your call.

Reply:
Yes. We should adhere to the contract. The contract is more lenient than the current implementation. It just requires negative, positive, or zero at the appropriate places, not -1, 0, or +1 specifically.